src/VectorizeLoops.cpp

```diff
@@ -1093,6 +1093,12 @@ class VectorSubs : public IRMutator {
                         reduce_op = VectorReduce::Or;
                     }
                 }
+            } else if (const Call *call_op = store->value.as<Call>()) {
+                if (call_op->is_intrinsic(Call::saturating_add)) {
```

Comment:
We should probably run `find_intrinsics` before `vectorize_loops` in lowering, otherwise this will not work if people write saturating add as a pattern they expect to match rather than using the (currently `Halide::Internal`) intrinsic.

This is a fairly big change to make, because the simplifier and other lowering passes don't understand `saturating_add` (yet? @rootjalex @abadams).

Reply:
That sounds reasonable, I will try to move the `find_intrinsics` earlier in the pipeline and I will the tests.

Reply:
@dsharletg I placed the `find_intrisics` pass right before `vectorize_loops`, thanks for the suggestion (b880a61b4b23faa8387193a21ceb15522b6afdf9)